content.js

```diff
@@ -1,16 +1,25 @@
 const myAudio = new Audio(chrome.runtime.getURL("ding.mp3"));
 
-const entireBody = document.body.innerText;
-if(entireBody.indexOf("No delivery windows available") != -1) {
-  console.log('no avail');
-  console.log('waiting 30 seconds');
-  setTimeout(() => {
-    window.location.reload();
-  }, 30000);
-}
-else {
-  console.log('bing!');
-  // get the body out for next time. pick slots automatically
-  console.log(document.body.innerHTML);
-  myAudio.play();
+const init = () => {
+  const entireBody = document.body.innerText;
+  // So it can match one of those messages: 
+    // - "No delivery windows available", 
+    // - "No attended delivery windows are available"
+    // - "No doorstep delivery windows are available";   
+  if (entireBody.search(/No(.*)delivery windows(.*)are available/) !== -1) {
+    console.log('no avail');
+    console.log('waiting 30 seconds');
+    setTimeout(() => {
+      window.location.reload();
+    }, 30000);
+  }
+  else {
+    console.log('bing!');
+    // get the body out for next time. pick slots automatically
+    console.log(document.body.innerHTML);
+    myAudio.play();
+  }
 }
+
+// Set a timeout to let client to render the page
+setTimeout(init, 3000);
```

Comment:
Oh, I see. I misunderstood what Chrome said here https://developer.chrome.com/extensions/content_scripts#run_time

Looks like content scripts run sometime after DOM ready but before `window.onload`. 

Now I also realized it's possible amazon is doing clientside rendering and injection. So waiting 3 seconds seems reasonable.

Couldn't we just continuously check the body then for the text in a tighter loop so we can alert right when it shows up though?